cpp/src/community/k_truss_impl.cuh

```diff
@@ -41,8 +42,175 @@
 #include <thrust/transform.h>
 #include <thrust/tuple.h>
 
+#include <chrono>
+using namespace std::chrono;
+
 namespace cugraph {
 
+template <typename vertex_t, typename edge_t, bool multi_gpu>
+void order_edge_based_on_dodg(raft::handle_t const& handle,
+                              graph_view_t<vertex_t, edge_t, false, multi_gpu>& graph_view,
+                              raft::device_span<vertex_t> edgelist_srcs,
+                              raft::device_span<vertex_t> edgelist_dsts)
```

Comment:
What are you doing in this function?

Reply:
Essentially this function is ordering the edges obtained from nbr_intersection (On the undirected graph) based on the DODG edges

Reply:
Once we identify weak edges, the next step is to retrieve the triangles incident to those weak edges and this is done with nbr_intersection (on the symmetric graph). Once we have the three endpoints of the triangle, the next step is to find the direction that matches the edges in the DODG and this is what that function does